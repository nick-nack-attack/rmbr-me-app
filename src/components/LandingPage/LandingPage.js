import React, { Component } from 'react';
import never_forget from '../../assets/never_forget.png'
import listening_skills from '../../assets/listening_skills.png'
import have_better from '../../assets/have_better.png'
import { Section } from '../Utils/Utils'

export default class LandingPage extends Component {

    handleSignUpNowClick = () => {
        const { history } = this.props;
        const destination = '/sign-up';
        history.push(destination);
    }

    render() {
        return (
            <Section
                className='landing_page_section'
            >
                <header
                    className='landing_page_header'
                >
                    <h2>No more forgetting or regretting</h2>
                </header>
                <section>
                    <img
                        src={listening_skills}
                        alt='lightning bolt listening to a comment'
                    />
                    <header>
                        <h3>Improve your listening skills</h3>
                    </header>
                    <p>rmbrme helps you remember things about your friends and family so you can be more engaged in their lives.</p>
                </section>
                <section>
                    <img
                        src={never_forget}
                        alt='lightning bolt remembering a comment'
                    />
                    <header>
                        <h3>Never forget a detail</h3>
                    </header>
                    <p>Stay up-to-date on the people who matter to you.</p>
                </section>
                <section>
                    <img
                        src={have_better}
                        alt='lightning bolt remembering comment'
                    />
                    <header>
                            <h3>Have better relationships</h3>
                    </header>
                    <p>People will appreciate you more as you remember things they've said.</p>
                </section>
                <section>
                    <header
                        className='sign_up_cta'
                    >
                        <h3>No more forgetting or regretting</h3>
                    </header>
                    <button
                        onClick={() => this.handleSignUpNowClick()}
                        className='sign_up_button'
                    >
                        Sign Up Now
                    </button>
                </section>
            </Section>
        );
    };
};