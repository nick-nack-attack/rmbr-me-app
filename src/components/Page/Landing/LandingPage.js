// this component explains the functionality of the app
import React, { Component } from 'react';

// utils
import { Section } from '../../Utils/Utils';

// assets & style
import never_forget from '../../../assets/never_forget.png';
import listening_skills from '../../../assets/listening_skills.png';
import have_better from '../../../assets/have_better.png';
import '../Page.scss';

export default class LandingPage extends Component {

    handleSignUpNowClick = () => {
        const { history } = this.props;
        const destination = '/sign-up';
        history.push(destination);
    }

    render() {
        return (
            <div className="landing-page-div">
                <Section className="landing-page-section">
                    <img src={listening_skills} alt='lightning bolt listening to a comment'/>
                    <h3>Improve your listening skills</h3>
                    <p>rmbrme helps you remember things about your friends and family so you can be more engaged in their lives.</p>
                </Section>
                <Section className="landing-page-section">
                    <img src={never_forget} alt='lightning bolt remembering a comment'/>
                    <h3>Never forget a detail</h3>
                    <p>Stay up-to-date on the people who matter to you.</p>
                </Section>
                <Section className="landing-page-section">
                    <img src={have_better} alt='lightning bolt remembering comment'/>
                    <h3>Have better relationships</h3>
                    <p>People will appreciate you more as you remember things they've said.</p>
                </Section>
                <Section className="landing-page-section last-section">
                    <div className="cta-div">
                        <h3>No more forgetting or regretting</h3>
                            <button
                                onClick={() => this.handleSignUpNowClick()}
                                className='sign_up_button'
                            >
                                Sign Up Now
                            </button>
                    </div>
                </Section>
            </div>
        );
    };
};