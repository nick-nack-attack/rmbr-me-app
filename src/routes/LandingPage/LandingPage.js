import React, { Component } from 'react';

class LandingPage extends Component {

    render() {
        console.log('landing page is running...');
        return (

            <div>
                <header>
                    <h1>rmbrme</h1>
                    <h2>be a better buddy and bro</h2>
                </header>
                <section>
                <header>
                    <h3>Improve your listening skills</h3>
                </header>
                    <p>[ <em>placeholder for photo showing person with big ears</em> ]</p>
                    <p>rmbrme helps you remember things about your friends and family so you can be more engaged in their lives.</p>
            </section>
            <section>
                <header>
                    <h3>Never forget a detail</h3>
                </header>
                    <p>[ <em>placeholder for photo showing person with laser eyes</em> ]</p>
                    <p>Stay up-to-date on the people you love or like a lot.</p>
            </section>
            <section>
                <header>
                    <h3>Have better relationships</h3>
                </header>
                    <p>[ <em>placeholder for photo showing person hugging a bunch of people</em> ]</p>
                    <p>People will appreciate you more as you remember more about them.</p>
            </section>
            <section>
                <header>
                    <h3>Start remembering now!</h3>
                </header>
                <form>
                    <div>
                        <label htmlFor="first-name">First name</label>
                        <input placeholder="First Name" type="text" name="first-name" id="first-name"/>
                    </div>
                    <div>
                        <label for="username">Email</label>
                        <input type="text" name="username" id="username"/>
                    </div>
                    <div>
                        <label for="password">Password</label>
                        <input type="password" name="password" id="password"/>
                    </div>
                    <div>
                        <button type="submit">Get started</button>
                    </div>
                </form>
            </section>
            </div>

        )

    }

}

export default LandingPage
