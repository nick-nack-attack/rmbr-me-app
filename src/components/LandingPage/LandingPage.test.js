import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LandingPage from "./LandingPage";

describe(`LandingPage component`, () => {

    it('renders LandingPage by default', () => {
        const wrapper = shallow(<LandingPage/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    
})