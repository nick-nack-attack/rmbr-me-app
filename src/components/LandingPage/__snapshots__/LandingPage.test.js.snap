// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders LandingPage by default 1`] = `
<Section
  className="landing_page_section"
>
  <header
    className="landing_page_header"
  >
    <h1>
      rmbrme
    </h1>
    <h2>
      be a better buddy
    </h2>
  </header>
  <section>
    <img
      src="listening_skills.png"
    />
    <header>
      <h3>
        Improve your listening skills
      </h3>
    </header>
    <p>
      rmbrme helps you remember things about your friends and family so you can be more engaged in their lives.
    </p>
  </section>
  <section>
    <img
      src="never_forget.png"
    />
    <header>
      <h3>
        Never forget a detail
      </h3>
    </header>
    <p>
      Stay up-to-date on the people you love or like a lot.
    </p>
  </section>
  <section>
    <img
      src="have_better.png"
    />
    <header>
      <h3>
        Have better relationships
      </h3>
    </header>
    <p>
      People will appreciate you more as you remember more about them.
    </p>
  </section>
  <section>
    <header
      className="sign_up_cta"
    >
      <h3>
        No more forgetting or regretting
      </h3>
    </header>
    <button
      className="sign_up_button"
      onClick={[Function]}
    >
      Sign Up Now
    </button>
  </section>
</Section>
`;
